### DISCUSSION TITLE
Yocto-dev mpfs-dev-cli-icicle-kit-es on SD card kernel panic

### DISCUSSION THREAD

**Discussion by eric-riscv** [polarfire-soc/polarfire-soc-documentation#174] [Linux build systems for PolarFire SoC] [answered]
Ubuntu 22.04 x64
repo init -u https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp.git -b master -m tools/manifests/riscv-yocto.xml

Build result is:
-rw-r--r-- 2 eric eric 1569692622 Aug  9 00:06 mpfs-dev-cli-icicle-kit-es-20220809044616.rootfs.wic.gz

Usbimager copied to:
SanDisk 256GB Extreme PRO SDXC UHS-I Memory Card - C10, U3, V30, 4K UHD, SD Card - SDSDXXD-256G-GN4IN
Said copy was success.

Inserted sd card.
Booted on icicle kit.

Failure was:

[    5.444432] Run /sbin/init as init process
[    5.570545] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    5.585463] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 9: comm init: Directory block failed checksum
[    5.601019] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    5.615880] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 9: comm init: Directory block failed checksum
/sbin/init: erro[    5.628538] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
r while loading [    5.637601] CPU: 1 PID: 1 Comm: init Not tainted 5.15.32-linux4microchip+fpga-2022.05 #1
shared libraries[    5.647062] Hardware name: Microchip PolarFire-SoC Icicle Kit (DT)
: libacl.so.1: c[    5.654607] Call Trace:
annot open share[    5.658431] [<ffffffff80004820>] dump_backtrace+0x1c/0x24
d object file: E[    5.665221] [<ffffffff804c2c7c>] show_stack+0x2c/0x38
rror 74
[    5.671651] [<ffffffff804c597e>] dump_stack_lvl+0x40/0x58
[    5.677823] [<ffffffff804c59aa>] dump_stack+0x14/0x1c
[    5.682927] [<ffffffff804c2df6>] panic+0xe8/0x292
[    5.687674] [<ffffffff8000d8fa>] do_exit+0x6fc/0x74a
[    5.692692] [<ffffffff8000e57a>] do_group_exit+0x2a/0x7e
[    5.698057] [<ffffffff8000e5e6>] __wake_up_parent+0x0/0x20
[    5.703603] [<ffffffff80002ffe>] ret_from_syscall+0x0/0x2
[    5.709055] SMP: stopping secondary CPUs
[    5.713032] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00 ]---

====

Full /dev/ttyUSB1 log

[7.685027] HSS_Boot_PMPSetupHandler(): Hart1 setup complete

U-Boot 2022.01 (Jan 10 2022 - 18:46:34 +0000)

CPU:   rv64imafdc
Model: Microchip PolarFire-SoC Icicle Kit
DRAM:  1.8 GiB
MMC:   mmc@20008000: 0
Loading Environment from nowhere... OK
In:    serial@20100000
Out:   serial@20100000
Err:   serial@20100000
Net:   eth0: ethernet@20112000
Hit any key to stop autoboot:  2  1  0 
Trying to boot from QSPI...
spi_nand spi-nand@0: unknown raw ID 1075789660
Could not find a valid device for spi-nand0
List of MTD devices:
No MTD device found
Trying to boot from eMMC/SD...
1326 bytes read in 19 ms (67.4 KiB/s)
## Executing script at 1008100000
4346428 bytes read in 206 ms (20.1 MiB/s)
## Loading kernel from FIT Image at 1008300000 ...
   Using 'conf-microchip_mpfs-icicle-kit.dtb' configuration
   Trying 'kernel-1' kernel subimage
     Description:  Linux kernel
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x10083000fc
     Data Size:    4328359 Bytes = 4.1 MiB
     Architecture: RISC-V
     OS:           Linux
     Load Address: 0x1000200000
     Entry Point:  0x1000200000
     Hash algo:    sha256
     Hash value:   f8af4c4ad5d8817c083dda0f7e230ddc00c790cfab2e05117144307bb577180b
   Verifying Hash Integrity ... sha256+ OK
## Loading fdt from FIT Image at 1008300000 ...
   Using 'conf-microchip_mpfs-icicle-kit.dtb' configuration
   Trying 'fdt-microchip_mpfs-icicle-kit.dtb' fdt subimage
     Description:  Flattened Device Tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x1008720dcc
     Data Size:    16068 Bytes = 15.7 KiB
     Architecture: RISC-V
     Load Address: 0x1002200000
     Hash algo:    sha256
     Hash value:   695ca6523b2cc3053c8426bf550e2d918363aa2205cad73e1650162295e1d044
   Verifying Hash Integrity ... sha256+ OK
   Loading fdt from 0x1008720dcc to 0x1002200000
   Booting using the fdt blob at 0x1002200000
   Uncompressing Kernel Image
   Loading Device Tree to 000000100fff9000, end 000000100ffffec3 ... OK

Starting kernel ...

[    0.000000] Linux version 5.15.32-linux4microchip+fpga-2022.05 (oe-user@oe-host) (riscv64-oe-linux-gcc (GCC) 11.2.0, GNU ld (GNU Binutils) 2.37.20210721) #1 SMP Thu May 26 12:32:36 UTC 2022
[    0.000000] OF: fdt: Ignoring memory range 0x1000000000 - 0x1000200000
[    0.000000] Machine model: Microchip PolarFire-SoC Icicle Kit
[    0.000000] earlycon: sbi0 at I/O port 0x0 (options '')
[    0.000000] printk: bootconsole [sbi0] enabled
[    0.000000] efi: UEFI not found.
[    0.000000] Reserved memory: created DMA memory pool at 0x0000000080000000, size 32 MiB
[    0.000000] OF: reserved mem: initialized node buffer@80000000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x00000000c0000000, size 128 MiB
[    0.000000] OF: reserved mem: initialized node buffer@c0000000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x00000000d0000000, size 128 MiB
[    0.000000] OF: reserved mem: initialized node buffer@d0000000, compatible id shared-dma-pool
[    0.000000] Zone ranges:
[    0.000000]   DMA32    empty
[    0.000000]   Normal   [mem 0x0000001000200000-0x0000001075ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000001000200000-0x0000001075ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000001000200000-0x0000001075ffffff]
[    0.000000] SBI specification v0.3 detected
[    0.000000] SBI implementation ID=0x1 Version=0x9
[    0.000000] SBI TIME extension detected
[    0.000000] SBI IPI extension detected
[    0.000000] SBI RFENCE extension detected
[    0.000000] SBI v0.2 HSM extension detected
[    0.000000] CPU with hartid=0 is not available
[    0.000000] CPU with hartid=0 is not available
[    0.000000] riscv: ISA extensions acdfim
[    0.000000] riscv: ELF capabilities acdfim
[    0.000000] percpu: Embedded 17 pages/cpu s30248 r8192 d31192 u69632
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 476215
[    0.000000] Kernel command line: earlycon=sbi root=/dev/mmcblk0p3 rootwait uio_pdrv_genirq.of_id=generic-uio
[    0.000000] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: Cannot allocate buffer
[    0.000000] Memory: 1884988K/1931264K available (4923K kernel code, 4813K rwdata, 2048K rodata, 2133K init, 287K bss, 46276K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: 	RCU event tracing is enabled.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=8 to nr_cpu_ids=4.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] CPU with hartid=0 is not available
[    0.000000] riscv-intc: unable to find hart id for /cpus/cpu@0/interrupt-controller
[    0.000000] riscv-intc: 64 local interrupts mapped
[    0.000000] plic: interrupt-controller@c000000: mapped 186 interrupts with 4 handlers for 9 contexts.
[    0.000000] random: get_random_bytes called from start_kernel+0x4c0/0x6ca with crng_init=0
[    0.000000] riscv_timer_init_dt: Registering clocksource cpuid [0] hartid [4]
[    0.000000] clocksource: riscv_clocksource: mask: 0xffffffffffffffff max_cycles: 0x1d854df40, max_idle_ns: 3526361616960 ns
[    0.000003] sched_clock: 64 bits at 1000kHz, resolution 1000ns, wraps every 2199023255500ns
[    0.000400] Console: colour dummy device 80x25
[    0.000495] printk: console [tty0] enabled
[    0.000583] printk: bootconsole [sbi0] disabled
[    0.000754] Calibrating delay loop (skipped), value calculated using timer frequency.. 2.00 BogoMIPS (lpj=4000)
[    0.000886] pid_max: default: 32768 minimum: 301
[    0.001275] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.001501] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.005389] ASID allocator disabled
[    0.005687] rcu: Hierarchical SRCU implementation.
[    0.006239] EFI services will not be available.
[    0.007340] smp: Bringing up secondary CPUs ...
[    0.011632] smp: Brought up 1 node, 4 CPUs
[    0.013060] devtmpfs: initialized
[    0.018857] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.018998] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.020501] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.049905] vgaarb: loaded
[    0.050505] SCSI subsystem initialized
[    0.051214] usbcore: registered new interface driver usbfs
[    0.051342] usbcore: registered new interface driver hub
[    0.051448] usbcore: registered new device driver usb
[    0.051772] pps_core: LinuxPPS API ver. 1 registered
[    0.051848] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.051976] PTP clock support registered
[    0.053356] clocksource: Switched to clocksource riscv_clocksource
[    0.065997] NET: Registered PF_INET protocol family
[    0.067433] IP idents hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.072915] tcp_listen_portaddr_hash hash table entries: 1024 (order: 3, 40960 bytes, linear)
[    0.073286] TCP established hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.074355] TCP bind hash table entries: 16384 (order: 7, 524288 bytes, linear)
[    0.077702] TCP: Hash tables configured (established 16384 bind 16384)
[    0.078620] UDP hash table entries: 1024 (order: 4, 98304 bytes, linear)
[    0.079307] UDP-Lite hash table entries: 1024 (order: 4, 98304 bytes, linear)
[    0.080284] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.080410] PCI: CLS 0 bytes, default 64
[    0.082114] workingset: timestamp_bits=62 max_order=19 bucket_order=0
[    0.093725] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 250)
[    0.093845] io scheduler mq-deadline registered
[    0.093907] io scheduler kyber registered
[    0.097538] irq: Cannot allocate irq_descs @ IRQ45, assuming pre-allocated
[    0.098186] microchip,mpfs-gpio 20122000.gpio: Microchip MPFS GPIO registered 32 GPIOs
[    0.098677] mchp-core-pwm 41000000.pwm: Successfully registered Microchip corePWM
[    0.099192] microchip-pcie 2000000000.pcie: host bridge /soc/pcie@2000000000 ranges:
[    0.099326] microchip-pcie 2000000000.pcie:      MEM 0x2008000000..0x2087ffffff -> 0x0008000000
[    0.153482] microchip-pcie 2000000000.pcie: axi read request error
[    0.153586] microchip-pcie 2000000000.pcie: axi read timeout
[    0.281483] microchip-pcie 2000000000.pcie: axi write request error
[    0.537472] microchip-pcie 2000000000.pcie: axi write request error
[    0.537594] microchip-pcie 2000000000.pcie: ECAM at [mem 0x2000000000-0x2007ffffff] for [bus 00-7f]
[    0.537905] microchip-pcie 2000000000.pcie: PCI host bridge to bus 0000:00
[    0.537991] pci_bus 0000:00: root bus resource [bus 00-7f]
[    0.538065] pci_bus 0000:00: root bus resource [mem 0x2008000000-0x2087ffffff] (bus address [0x08000000-0x87ffffff])
[    0.538227] pci 0000:00:00.0: [11aa:1556] type 01 class 0x000604
[    0.538310] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x7fffffff 64bit pref]
[    0.538512] pci 0000:00:00.0: supports D1 D2
[    0.538564] pci 0000:00:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    0.540620] pci 0000:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    0.592088] pci_bus 0000:01: busn_res: [bus 01-7f] end is updated to 01
[    0.592213] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000 64bit pref]
[    0.592315] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000 64bit pref]
[    0.592418] pci 0000:00:00.0: BAR 8: assigned [mem 0x2008000000-0x20081fffff]
[    0.592488] pci 0000:00:00.0: BAR 9: assigned [mem 0x2008200000-0x20083fffff 64bit pref]
[    0.592572] pci 0000:00:00.0: BAR 7: no space for [io  size 0x1000]
[    0.592644] pci 0000:00:00.0: BAR 7: failed to assign [io  size 0x1000]
[    0.592722] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.592779] pci 0000:00:00.0:   bridge window [mem 0x2008000000-0x20081fffff]
[    0.592846] pci 0000:00:00.0:   bridge window [mem 0x2008200000-0x20083fffff 64bit pref]
[    0.667698] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
[    0.670526] 20100000.serial: ttyS1 at MMIO 0x20100000 (irq = 4, base_baud = 9375000) is a 16550A
[    1.449904] printk: console [ttyS1] enabled
[    1.455474] 20102000.serial: ttyS2 at MMIO 0x20102000 (irq = 6, base_baud = 9375000) is a 16550A
[    1.465538] 20104000.serial: ttyS3 at MMIO 0x20104000 (irq = 7, base_baud = 9375000) is a 16550A
[    1.475534] 20106000.serial: ttyS0 at MMIO 0x20106000 (irq = 8, base_baud = 9375000) is a 16550A
[    1.495482] loop: module loaded
[    1.652962] u-dma-buf udmabuf-ddr-c0: driver version = 3.2.4
[    1.658796] u-dma-buf udmabuf-ddr-c0: major number   = 249
[    1.664413] u-dma-buf udmabuf-ddr-c0: minor number   = 0
[    1.669806] u-dma-buf udmabuf-ddr-c0: phys address   = 0x0000000080000000
[    1.676677] u-dma-buf udmabuf-ddr-c0: buffer size    = 33554432
[    1.682680] u-dma-buf udmabuf0: driver installed.
[    4.706458] u-dma-buf udmabuf-ddr-nc0: driver version = 3.2.4
[    4.712319] u-dma-buf udmabuf-ddr-nc0: major number   = 249
[    4.717983] u-dma-buf udmabuf-ddr-nc0: minor number   = 1
[    4.723453] u-dma-buf udmabuf-ddr-nc0: phys address   = 0x00000000c0000000
[    4.730425] u-dma-buf udmabuf-ddr-nc0: buffer size    = 134217728
[    4.736594] u-dma-buf udmabuf1: driver installed.
[    4.808088] u-dma-buf udmabuf-ddr-nc-wcb0: driver version = 3.2.4
[    4.814281] u-dma-buf udmabuf-ddr-nc-wcb0: major number   = 249
[    4.820281] u-dma-buf udmabuf-ddr-nc-wcb0: minor number   = 2
[    4.826108] u-dma-buf udmabuf-ddr-nc-wcb0: phys address   = 0x00000000d0000000
[    4.833438] u-dma-buf udmabuf-ddr-nc-wcb0: buffer size    = 134217728
[    4.839949] u-dma-buf udmabuf2: driver installed.
[    4.847689] microchip-corespi 20108000.spi: Registered SPI controller 0
[    4.855201] microchip-corespi 20109000.spi: Registered SPI controller 1
[    4.894585] macb 20110000.ethernet eth0: Cadence GEM rev 0x0107010c at 0x20110000 irq 16 (00:04:a3:e9:25:ce)
[    4.909211] macb 20112000.ethernet eth1: Cadence GEM rev 0x0107010c at 0x20112000 irq 22 (00:04:a3:e9:25:cd)
[    4.920741] can-uio 2010c000.can: registered device as can@2010c000
[    4.927690] mss-dma-uio 60020000.dma-controller: registered device as dma-controller@60020000
[    4.936448] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    4.943084] ehci-pci: EHCI PCI platform driver
[    4.947681] ehci-platform: EHCI generic platform driver
[    4.953165] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    4.959441] ohci-pci: OHCI PCI platform driver
[    4.964007] ohci-platform: OHCI generic platform driver
[    4.970601] musb-hdrc musb-hdrc.1.auto: MUSB HDRC host driver
[    4.976467] musb-hdrc musb-hdrc.1.auto: new USB bus registered, assigned bus number 1
[    4.985411] hub 1-0:1.0: USB hub found
[    4.989276] hub 1-0:1.0: 1 port detected
[    4.993727] mpfs-musb 20201000.usb: Registered MPFS MUSB driver
[    5.000330] mpfs_rtc 20124000.rtc: prescaler set to: 0xF423F 
[    5.006616] mpfs_rtc 20124000.rtc: registered as rtc0
[    5.011799] mpfs_rtc 20124000.rtc: setting system clock to 1970-01-01T00:00:00 UTC (0)
[    5.019966] i2c_dev: i2c /dev entries driver
[    5.024962] microchip-corei2c 2010a000.i2c: Microchip I2C Probe Complete
[    5.032480] microchip-corei2c 2010b000.i2c: Microchip I2C Probe Complete
[    5.039780] microchip-corei2c 44000000.i2c: Microchip I2C Probe Complete
[    5.046963] sdhci: Secure Digital Host Controller Interface driver
[    5.053245] sdhci: Copyright(c) Pierre Ossman
[    5.057733] sdhci-pltfm: SDHCI platform and OF driver helper
[    5.063917] usbcore: registered new interface driver usbhid
[    5.069602] usbhid: USB HID core driver
[    5.073771] mpfs-mailbox 37020000.mailbox: Registered MPFS mailbox controller driver
[    5.083389] NET: Registered PF_INET6 protocol family
[    5.090958] Segment Routing with IPv6
[    5.094788] In-situ OAM (IOAM) with IPv6
[    5.095765] mmc0: SDHCI controller on 20008000.mmc [20008000.mmc] using ADMA 64-bit
[    5.098926] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    5.113631] NET: Registered PF_PACKET protocol family
[    5.119323] mpfs-sys-controller syscontroller: Registered MPFS system controller
[    5.128574] mpfs-rng mpfs-rng: Registered MPFS hwrng
[    5.133974] mpfs-generic-service mpfs-generic-service: Registered MPFS generic service
[    5.142282] of_cfs_init
[    5.144980] of_cfs_init: OK
[    5.149403] Waiting for root device /dev/mmcblk0p3...
[    5.149929] random: fast init done
[    5.170129] sdhci-cdns 20008000.mmc: no tuning point found
[    5.171438] random: crng init done
[    5.175724] mmc0: tuning execution failed: -5
[    5.175752] mmc0: error -5 whilst initialising SD card
[    5.274592] sdhci-cdns 20008000.mmc: no tuning point found
[    5.280170] mmc0: tuning execution failed: -5
[    5.349673] mmc0: new high speed SDXC card at address aaaa
[    5.357416] mmcblk0: mmc0:aaaa SR256 238 GiB 
[    5.369646] Alternate GPT is invalid, using primary GPT.
[    5.375143]  mmcblk0: p1 p2 p3
[    5.406432] EXT4-fs (mmcblk0p3): mounted filesystem with ordered data mode. Opts: (null). Quota mode: disabled.
[    5.416832] VFS: Mounted root (ext4 filesystem) readonly on device 179:3.
[    5.424796] devtmpfs: mounted
[    5.438263] Freeing unused kernel image (initmem) memory: 2132K
[    5.444432] Run /sbin/init as init process
[    5.570545] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    5.585463] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 9: comm init: Directory block failed checksum
[    5.601019] EXT4-fs warning (device mmcblk0p3): ext4_dirblock_csum_verify:376: inode #10943: comm init: No space for directory leaf checksum. Please run e2fsck -D.
[    5.615880] EXT4-fs error (device mmcblk0p3): ext4_dx_find_entry:1725: inode #10943: block 9: comm init: Directory block failed checksum
/sbin/init: erro[    5.628538] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
r while loading [    5.637601] CPU: 1 PID: 1 Comm: init Not tainted 5.15.32-linux4microchip+fpga-2022.05 #1
shared libraries[    5.647062] Hardware name: Microchip PolarFire-SoC Icicle Kit (DT)
: libacl.so.1: c[    5.654607] Call Trace:
annot open share[    5.658431] [<ffffffff80004820>] dump_backtrace+0x1c/0x24
d object file: E[    5.665221] [<ffffffff804c2c7c>] show_stack+0x2c/0x38
rror 74
[    5.671651] [<ffffffff804c597e>] dump_stack_lvl+0x40/0x58
[    5.677823] [<ffffffff804c59aa>] dump_stack+0x14/0x1c
[    5.682927] [<ffffffff804c2df6>] panic+0xe8/0x292
[    5.687674] [<ffffffff8000d8fa>] do_exit+0x6fc/0x74a
[    5.692692] [<ffffffff8000e57a>] do_group_exit+0x2a/0x7e
[    5.698057] [<ffffffff8000e5e6>] __wake_up_parent+0x0/0x20
[    5.703603] [<ffffffff80002ffe>] ret_from_syscall+0x0/0x2
[    5.709055] SMP: stopping secondary CPUs
[    5.713032] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00 ]---

**Comment by TommyMurphyTM1234** [comment, marked as answer]
Sounds similar, if not identical, to this issue?

https://github.com/orgs/polarfire-soc/discussions/35

Looks like, in some cases and for some reason, USBImager may be writing the `.wic.gz` file to the SD card instead of decompressing it on the fly and writing the actual `.wic` image to the SD card.

Could it be that in some installations of USBImager the option to decompress files on the fly is disabled, or maybe the detection of compressed files fails for some reason? I can't see anything obvious in the USBImager issue tracker (even the closed issues).

https://gitlab.com/bztsrc/usbimager/-/issues/?sort=created_date&state=closed

As such, it may be more reliable to manually decompress the `.wic.gz` file to extract the `.wic` image file, and then use USBImager (or maybe simply `dd` or similar?) to explicitly write the `.wic` image to the SD card rather than depending on USBImager to decompress on the fly?

**Reply by eric-riscv** [reply]
Copying the .wic (and NOT the .wic.gz file) using usbimager to the SD card worked perfectly from the yocto-dev build.

**Reply by TommyMurphyTM1234** [reply]
Glad that helped @eric-riscv.
Just curious, were you using Ubuntu on WSL (Windows Subsystem for Linux) by any chance - or was it a "regular" Ubuntu installation?

**Reply by eric-riscv** [reply]
Regular Ubuntu 22.04.  usbimager directly from website download, deb file.
